feature-libs/checkout/components/services/express-checkout.service.ts

```diff
@@ -283,17 +299,16 @@ export class ExpressCheckoutService {
             return of(false);
           }
         }
-      )
+      ),
+      distinctUntilChanged()
     );
   }
 
   public trySetDefaultCheckoutDetails(): Observable<boolean> {
     this.clearCheckoutService.resetCheckoutProcesses();
 
-    return combineLatest([this.deliveryModeSet$, this.paymentMethodSet$]).pipe(
-      map(([deliveryModeSet, paymentMethodSet]) =>
-        Boolean(deliveryModeSet && paymentMethodSet)
-      )
+    return this.paymentMethodSet$.pipe(
+      map((paymentMethodSet) => Boolean(paymentMethodSet))
```

Comment:
Why do we remove the deliveryMode?

Reply:
Ah, I think I see.  In the stream for `this.paymentMethodSet$`, deliveryModeSet is checked.